src/components/TodoList/TodoList.jsx

```diff
@@ -1,44 +1,167 @@
+        todos, todosCopy: todos,
+      }));
+  }
+
+  filteringTodos = () => {
+    this.setState(prevState => ({
+      todos: prevState.todosCopy.filter(
+        todo => todo.title && todo.title.includes(prevState.inputText),
+      ),
+    }));
+  };
+
+  handleSelect = (selectEvent) => {
+    const { value } = selectEvent.target;
+
+    this.setState({
+      selectValue: value,
+    });
+
+    this.setState((prevState) => {
+      if (value === 'active') {
+        return {
+          todos: prevState.todosCopy.filter(
+            todo => !todo.completed,
+          ),
+        };
+      }
+
+      if (value === 'completed') {
+        return {
+          todos: prevState.todosCopy.filter(
+            todo => todo.completed,
+          ),
+        };
+      }
+
+      return { todos: prevState.todosCopy };
+    });
+  }
+
+  handleChange = (changeEvent) => {
+    this.setState({
+      inputText: changeEvent.target.value,
+    });
+    this.filteringTodos();
+  };
+
+  shuffle = () => {
+    this.setState((prevState) => {
+      const shuffleTodos = prevState.todosCopy;
+      let currentIndex = shuffleTodos.length;
+      let temporaryValue;
+      let randomIndex;
+
+      while (currentIndex !== 0) {
+        randomIndex = Math.floor(Math.random() * currentIndex);
+        currentIndex -= 1;
+
+        temporaryValue = shuffleTodos[currentIndex];
+        shuffleTodos[currentIndex] = shuffleTodos[randomIndex];
+        shuffleTodos[randomIndex] = temporaryValue;
+      }
+
+      return {
+        todos: shuffleTodos,
+        selectValue: 'all',
+      };
+    });
+  }
+
+  render() {
+    const { inputText, selectValue } = this.state;
+
+    return (
+      <div className="TodoList">
+        <h2>Todos:</h2>
+
+        <div className="TodoList__list-container">
+          <div>
+            <input
+              type="text"
+              name="todos-filter"
+              onChange={this.handleChange}
+              value={inputText}
+              className="search-todos"
+            />
+            <select
+              value={selectValue}
+              name="action-filter"
+              id="action-filter"
+              onChange={this.handleSelect}
+              className="select-todos"
+            >
+              <option value="all">all</option>
+              <option value="active">active</option>
+              <option value="completed">completed</option>
+            </select>
+            <button
+              onClick={this.shuffle}
+              type="button"
+              className="Randomize"
+            >
+              Randomize
+            </button>
+          </div>
+          <ul className="TodoList__list">
+
+            {this.state.todos.map(todo => (
+              <li
+                key={todo.id}
+                className={
+                  todo.completed
+                    ? 'TodoList__item TodoList__item--checked'
+                    : 'TodoList__item TodoList__item--unchecked'
```

Comment:
в качестве рефакторинга, можно присваивать не два класса сразу, а один обоим, а второй по условию
```suggestion
                   className={`TodoList__item ${todo.completed === true
                  ? 'TodoList__item--checked'
                  : 'TodoList__item--unchecked'}`}
```